src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -0,0 +1,536 @@
+            (m_algo == STL && !one_of(out_prc, InferenceEngine::Precision::FP32)))) {
+        out_prc = InferenceEngine::Precision::FP32;
+    }
+    if (!out_prc.is_float() && !one_of(out_prc, InferenceEngine::Precision::I32, InferenceEngine::Precision::I64)) {
+        out_prc = InferenceEngine::Precision::I32;
+    }
+    m_output_prc = InferenceEngine::details::convertPrecision(out_prc);
+
+    addSupportedPrimDesc({{LayoutType::ncsp, shape_prc, m_const_inputs[SHAPE]},
+                          {LayoutType::ncsp, out_prc, m_const_inputs[MIN_VAL]},
+                          {LayoutType::ncsp, out_prc, m_const_inputs[MAX_VAL]}},
+                         {{LayoutType::ncsp, out_prc}},
+                         ref_any);
+}
+
+void RandomUniform::createPrimitive() {
+    if (m_const_inputs[MIN_VAL]) {
+        initEdgeValues(m_min_val, getParentEdgeAt(MIN_VAL)->getMemoryPtr()->getData(), m_output_prc);
+    }
+    if (m_const_inputs[MAX_VAL]) {
+        initEdgeValues(m_max_val, getParentEdgeAt(MAX_VAL)->getMemoryPtr()->getData(), m_output_prc);
+        evalRange();
+    }
+
+    if (m_algo == PHILOX) {
+#if defined(OPENVINO_ARCH_X86_64)
+        kernel::RandomUniformCompileParams jcp;
+
+        jcp.out_data_type = m_output_prc;
+
+        m_jit_kernel = kernel::JitKernel<kernel::RandomUniformCompileParams, kernel::RandomUniformCallArgs>::createInstance<kernel::RandomUniform>(jcp);
+
+        if (m_jit_kernel) {
+            if (auto selected_pd = getSelectedPrimitiveDescriptor()) {
+                using namespace dnnl::impl::cpu;
+                if (m_jit_kernel->getIsa() == x64::avx512_core) {
+                    selected_pd->setImplementationType(jit_avx512);
+                } else if (m_jit_kernel->getIsa() == x64::avx2) {
+                    selected_pd->setImplementationType(jit_avx2);
+                } else if (m_jit_kernel->getIsa() == x64::sse41) {
+                    selected_pd->setImplementationType(jit_sse42);
+                }
+            }
+        }
+#endif // OPENVINO_ARCH_X86_64
+    }
+
+    if (m_const_inputs[SHAPE]) {
+        Node::createPrimitive();
+    }
+}
+
+bool RandomUniform::needPrepareParams() const {
+    if (m_out_shape != getChildEdgeAt(0)->getMemoryPtr()->getShape().getStaticDims()) {
+        return true;
+    }
+    return false;
+}
+
+void RandomUniform::prepareParams() {
+    m_out_shape = getChildEdgeAt(0)->getMemoryPtr()->getShape().getStaticDims();
+    m_out_el_num = std::accumulate(m_out_shape.begin(), m_out_shape.end(), 1lu, std::multiplies<Dim>());
+
+    if (m_algo == PHILOX) {
+        m_skip_count = m_out_el_num * SKIP_CONST;
+
+        if (m_out_el_num < PHILOX_PARALLEL_EXECUTION_THRESHOLD) {
+            m_threads_num = 1;
+        } else {
+            m_threads_num = parallel_get_max_threads();
+        }
+        m_thread_params.resize(m_threads_num);
+
+        parallel_nt(m_threads_num, [&](const int ithr, const int nthr) {
```

Comment:
Why we need to have it on prepareParams stage? The preparation step seems to be very simple and should significantly affect execution time - so can be united with execute call for simplification.

Reply:
We have a couple of "fast" models where the RandomUniform is a quite small.  One of them took less than 1 μs and it is sensitive to such work.